openreview/tools.py

```diff
@@ -796,38 +797,23 @@ def replace_members_with_ids(client, group):
     :return: Group with the emails replaced by Profile ids
     :rtype: Group
     """
-    ids = []
-    emails = []
-
-    def classify_members(member):
-        if '@' in member:
-            try:
-                profile = client.get_profile(member.lower())
-                return 'ids', profile.id
-            except openreview.OpenReviewException as e:
-                if 'Profile Not Found' in e.args[0]:
-                    return 'emails', member.lower()
-                else:
-                    raise e
-        elif '~' in member:
-            profile = client.get_profile(member)
-            return 'ids', profile.id
-        else:
-            _group = client.get_group(member)
-            return 'ids', _group.id
+    updated_members = []
 
-    results = concurrent_requests(classify_members, group.members)
+    member_profiles = get_profiles(client, group.members, as_dict=True)
 
-    for key, member in results:
-        if key == 'ids':
-            ids.append(member)
-        elif key == 'emails':
-            emails.append(member)
+    for member, profile in member_profiles.items():
```

Comment:
could you start the iteration with `group.members` to make sure each member is replaced or not found error. 

```
for member in groups.members:
   profile = member_profiles.get(member)
   if profile:
          update_members.append(profile.id)
   else:
          if '@' in member:
                update_members.append(profile.id)
         else:
               raise openreview.OpenReviewException(f"Profile Not Found for {member}")
```

We also make sure they are replaced in the same order.

Reply:
Why do they need to be replaced in the same order?

Reply:
I think maintaining the order is better but I may be wrong.

Reply:
Well, they are now being replaced in order, so I think it's fine.